calcom/cal.com
# feat: Calendar Cache

## What does this PR do?

- Implements a basic third-party calendar cache that lasts for a minute (for v1)
- First call will be as slow as now, but subsequent calls should be nearly instant.
- Migrates gCal CalendarService from callbacks to async/awaits

## Possible downsides

- If a profile is a high traffic one, we might run into a race condition where the cache is de-synced with the actual availability and get a "Could not book the meeting." error.

## TODO

- [x] Implement a cron for cleanup expired caches
- [x] Implement an opt-in feature mechanism 
- [ ] Implement a way to bypass cache altogether (either via request header or search parameter)
- [x] Make the cache more aggressive (save full months, even if only a day was requested)

## Type of change




## How should this be tested?

- Run migrations
- Install Google Calendar
- Enable some calendars
- As an ADMIN, enable the calendar-cache feature flag
- Go to a booking page
- Check if CalendarCache table is being populated correctly
- Check availability works as normal

## Mandatory Tasks
